src/libopensc/card-entersafe.c

```diff
@@ -1135,22 +1155,47 @@ static int entersafe_write_rsa_key_factor(sc_card_t *card,
 		 sc_format_apdu(card,&apdu,SC_APDU_CASE_3_SHORT,0x22,0x01,0xB8);
 		 apdu.data=sbuff;
 		 apdu.lc=apdu.datalen=4;
-		 
+
 		 r=entersafe_transmit_apdu(card,&apdu,0,0,0,0);
 		 SC_TEST_RET(card->ctx, SC_LOG_DEBUG_NORMAL, r, "APDU transmit failed");
 		 SC_TEST_RET(card->ctx, SC_LOG_DEBUG_NORMAL, sc_check_sw(card, apdu.sw1, apdu.sw2),"Write prkey factor failed(MSE)");
 	}
 
 	{/* Write 'x'; */
-		u8 sbuff[SC_MAX_APDU_BUFFER_SIZE];
-
+		 u8 sbuff[SC_MAX_APDU_BUFFER_SIZE] = { 0 , };
 		 sc_format_apdu(card,&apdu,SC_APDU_CASE_3_SHORT,0x46,factor,0x00);
-
 		 memcpy(sbuff,data.data,data.len);
 		 entersafe_reverse_buffer(sbuff,data.len);
+/*
+ *  PK01C and PK13C smart card only support 1024 or 2048bit key .
+ *  Size of exponent1 exponent2 coefficient of RSA private key keep the same as size of prime1
+ *  So check factor is padded with zero or not
+ */
+		 switch(factor){
```

Comment:
Hi Frank

Yes, it will needed.

We must ensure that the length is 1024 or 2048, if not, we do padding using 0 

Thanks,